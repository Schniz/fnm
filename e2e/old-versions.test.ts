import { script } from "./shellcode/script.js"
import { Bash, Fish, PowerShell, WinCmd, Zsh } from "./shellcode/shells.js"
import testNodeVersion from "./shellcode/test-node-version.js"
import describe from "./describe.js"

for (const shell of [Bash, Zsh, Fish, PowerShell, WinCmd]) {
  describe(shell, () => {
    test(`download old Node.js 0.10.x`, async () => {
      await script(shell)
        .then(shell.env({}))
        .then(shell.call("fnm", ["install", "v0.10.36"]))
        .then(shell.call("fnm", ["use", "v0.10.36"]))
        .then(testNodeVersion(shell, "v0.10.36"))
        .takeSnapshot(shell)
        .execute(shell)
    })
  })
}
