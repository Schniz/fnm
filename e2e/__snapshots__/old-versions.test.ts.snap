// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bash download old Node.js 0.10.x: Bash 1`] = `
"set -e
eval "$(fnm env)"
fnm install v0.10.36
fnm use v0.10.36
if [ "$(node --version)" != "v0.10.36" ]; then
  echo "Expected node version to be v0.10.36. Got $(node --version)"
  exit 1
fi"
`;

exports[`Fish download old Node.js 0.10.x: Fish 1`] = `
"fnm env | source
fnm install v0.10.36
fnm use v0.10.36
set ____test____ (node --version)
if test "$____test____" != "v0.10.36"
  echo "Expected node version to be v0.10.36. Got $____test____"
  exit 1
end"
`;

exports[`PowerShell download old Node.js 0.10.x: PowerShell 1`] = `
"$ErrorActionPreference = "Stop"
fnm env | Out-String | Invoke-Expression
fnm install v0.10.36
fnm use v0.10.36
if ( "$(node --version)" -ne "v0.10.36" ) { echo "Expected node version to be v0.10.36. Got $(node --version)"; exit 1 }"
`;

exports[`Zsh download old Node.js 0.10.x: Zsh 1`] = `
"set -e
eval "$(fnm env)"
fnm install v0.10.36
fnm use v0.10.36
if [ "$(node --version)" != "v0.10.36" ]; then
  echo "Expected node version to be v0.10.36. Got $(node --version)"
  exit 1
fi"
`;
